vllm/distributed/kv_transfer/kv_connector/factory.py

```diff
@@ -59,6 +59,15 @@ def create_connector_v1(
                              f"but found {envs.VLLM_USE_V1=}")
 
         kv_transfer_config = config.kv_transfer_config
+        if not hasattr(kv_transfer_config, "_initialized_engine_id"):
```

Comment:
An alternative solution is to move this logic into `DPEngineCoreProc._init_dataparallel`, basically force regenerate the `engine_id` there instead of in the factory.

Reply:
Yes SGTM